File: 3803.count-residue-prefixes/solution.go
func residuePrefixes(s string) int {
    visited := make([]bool, 26)
    distinct := 0
    res := 0

    for i, ch := range s {
        if index := ch - 'a'; !visited[index] {
            visited[index] = true
            distinct++
        }
        if distinct == (i + 1) % 3 {
            res++
        }
    }

    return res
}

File: 3803.count-residue-prefixes/solution.kt
class Solution {
    fun residuePrefixes(s: String): Int {
        val letters = mutableSetOf<Char>()
        var res = 0

        s.forEachIndexed { i, ch ->
            letters.add(ch)
            if (letters.size == (i + 1) % 3) {
                res++
            }
        }

        return res
    }
}